circus-train-avro/src/test/java/com/hotels/bdp/circustrain/avro/transformation/AbstractAvroSerDeTransformationTest.java

```diff
@@ -0,0 +1,154 @@
+    assertThat(transformation.getTableLocation(), is("location"));
+  }
+
+  @Test
+  public void testMultipleReplicationsOverride() throws Exception {
+    EventTableReplication tableReplication = mockTableReplicaiton("overrideBaseUrl");
+    runLifeCycleSuccess(tableReplication);
+    assertThat("overrideBaseUrl", is(transformation.getAvroSchemaDestinationFolder()));
+    assertThat(EVENT_ID, is(transformation.getEventId()));
+    assertThat("location", is(transformation.getTableLocation()));
+    EventTableReplication tableReplication2 = mockTableReplicaiton("overrideBaseUrl2");
+    runLifeCycleSuccess(tableReplication2);
+    assertThat(transformation.getAvroSchemaDestinationFolder(), is("overrideBaseUrl2"));
+    assertThat(transformation.getEventId(), is(EVENT_ID));
+    assertThat(transformation.getTableLocation(), is("location"));
+  }
+
+  @Test
+  public void testMultipleReplicationsOverrideToNull() throws Exception {
+    EventTableReplication tableReplication = mockTableReplicaiton("overrideBaseUrl");
+    runLifeCycleSuccess(tableReplication);
+    assertThat("overrideBaseUrl", is(transformation.getAvroSchemaDestinationFolder()));
+    assertThat(EVENT_ID, is(transformation.getEventId()));
+    assertThat("location", is(transformation.getTableLocation()));
+    EventTableReplication tableReplication2 = mockTableReplicaiton(null);
+    runLifeCycleSuccess(tableReplication2);
+    assertThat(transformation.getAvroSchemaDestinationFolder(), is("default"));
+    assertThat(transformation.getEventId(), is(EVENT_ID));
+    assertThat(transformation.getTableLocation(), is("location"));
+  }
+
+  @Test
+  public void testOneReplicationsOverrideFailure() throws Exception {
+    EventTableReplication tableReplication = mockTableReplicaiton("overrideBaseUrl");
+    runLifeCycleFailure(tableReplication);
+    assertThat(transformation.getAvroSchemaDestinationFolder(), is("overrideBaseUrl"));
+    assertThat(transformation.getEventId(), is(EVENT_ID));
+    assertThat(transformation.getTableLocation(), is("location"));
+  }
+
+  @Test
+  public void testMultipleReplicationsOverrideFailure() throws Exception {
+    EventTableReplication tableReplication = mockTableReplicaiton("overrideBaseUrl");
+    runLifeCycleSuccess(tableReplication);
+    assertThat("overrideBaseUrl", is(transformation.getAvroSchemaDestinationFolder()));
+    assertThat(EVENT_ID, is(transformation.getEventId()));
+    assertThat("location", is(transformation.getTableLocation()));
+    EventTableReplication tableReplication2 = mockTableReplicaiton("overrideBaseUrl2");
+    runLifeCycleFailure(tableReplication2);
+    assertThat(transformation.getAvroSchemaDestinationFolder(), is("overrideBaseUrl2"));
+    assertThat(transformation.getEventId(), is(EVENT_ID));
+    assertThat(transformation.getTableLocation(), is("location"));
+  }
+
+  @Test
+  public void testMultipleReplicationsOverrideToNullFailure() throws Exception {
```

Comment:
Same here. testMultipleReplicationsOverrideFailureWithNullBaseURL() will be a better name.